sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2OboAuthorizedClientRepository.java

```diff
@@ -130,4 +161,15 @@ private String interceptAuthorizationUri(String authorizationUri) {
         }
         return null;
     }
+
+    void replyForbiddenWithWwwAuthenticateHeader(HttpServletResponse response,
+                                                 MsalInteractionRequiredException exception) {
```

Comment:
Can we get `response` by `RequestContextHolder` in this method instead of pass it in?

Reply:
Maybe `ExceptionTranslationFilter` is better than `WebAsyncManagerIntegrationFilter`, `ExceptionTranslationFilter` handles AccessDeniedException and AuthenticationException. So `ExceptionTranslationFilter` always exists. It handles exceptions like our filter, so it’s better to add `ExceptionTranslationFilter`.

About `replyForbiddenWithWwwAuthenticateHeader`. Your idea is better, I have changed the code.